File: src-tauri/service/src/models/order.rs
use sea_orm::FromQueryResult;
use serde::{Deserialize, Serialize};
use crate::{NewOrderItem,UpdateOrderItem};

#[derive(Deserialize, Serialize, Debug, PartialEq, FromQueryResult)]
pub struct SelectOrders {
    pub id: String,
    pub created_at: String,
    pub client_id: String,
    pub full_name: String,
    pub status: String,
    pub identifier: String,
    pub products: i64,
    pub total: f64,
}

#[derive(Deserialize, Serialize, Debug, PartialEq, FromQueryResult)]
pub struct SelectOrderDetails {
    pub id: String,
    pub created_at: String,
    pub full_name: String,
    pub identifier: String,
    pub address: Option<String>,
    pub phone_number: Option<String>,
    pub email: Option<String>,
    pub status: String,
    pub total: f64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct NewOrder {
    pub client_id: String,
    pub status: String,
    pub items: Vec<NewOrderItem>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateOrder {
    pub id: String,
    pub client_id: String,
    pub status: String,
    pub items: Vec<UpdateOrderItem>,
}

File: types/order.ts
import type { ORDER_STATUSES } from "~/consts/status";

export interface ListOrderT {
  id: string;
  status: (typeof ORDER_STATUSES)[number];
  client_id: string;
  full_name?: string;
  products?: number;
  total?: number;
  created_at?: string;
  identifier: string;
}

export interface OrderForUpdateT {
  id: string;
  client_id: string;
  full_name: string;
  created_at: string;
  identifier?: string;
  status: (typeof ORDER_STATUSES)[number];
  items: {
    id: string;
    inventory_id: string;
    name: string;
    product_id: string;
    quantity: number;
    price: number;
  }[];
}

export interface OrderProductsPreviewT {
  price: number;
  name: string;
  quantity: number;
}
